notifications-kibana-plugin/public/pages/CreateChannel/CreateChannel.tsx

```diff
@@ -0,0 +1,273 @@
+  setInputErrors: (errors: CreateChannelInputErrorsType) => void;
+} | null>(null);
+
+export type HeaderType = { key: string; value: string };
+
+export function CreateChannel(props: CreateChannelsProps) {
+  const coreContext = useContext(CoreServicesContext)!;
+  const id = props.match.params.id;
+  const prevURL =
+    props.edit && queryString.parse(props.location.search).from === 'details'
+      ? `#${ROUTES.CHANNEL_DETAILS}/${id}`
+      : `#${ROUTES.CHANNELS}`;
+
+  const [name, setName] = useState('');
+  const [description, setDescription] = useState('');
+
+  const channelTypeOptions: Array<EuiSuperSelectOption<
+    keyof typeof CHANNEL_TYPE
+  >> = Object.entries(CHANNEL_TYPE).map(([key, value]) => ({
+    value: key as keyof typeof CHANNEL_TYPE,
+    inputDisplay: value,
+  }));
+  const [channelType, setChannelType] = useState(channelTypeOptions[0].value);
+
+  const [slackWebhook, setSlackWebhook] = useState('');
+
+  const [
+    headerFooterCheckboxIdToSelectedMap,
+    setHeaderFooterCheckboxIdToSelectedMap,
+  ] = useState<{
+    [x: string]: boolean;
+  }>({});
+  const [emailHeader, setEmailHeader] = useState('');
+  const [emailFooter, setEmailFooter] = useState('');
+  const [sender, setSender] = useState('');
+  const [
+    selectedRecipientGroupOptions,
+    setSelectedRecipientGroupOptions,
+  ] = useState<Array<EuiComboBoxOptionOption<string>>>([]);
+
+  const [webhookTypeIdSelected, setWebhookTypeIdSelected] = useState<
+    keyof typeof CUSTOM_WEBHOOK_ENDPOINT_TYPE
+  >('WEBHOOK_URL');
+  const [webhookURL, setWebhookURL] = useState('');
+  const [customURLHost, setCustomURLHost] = useState('');
+  const [customURLPort, setCustomURLPort] = useState('');
+  const [customURLPath, setCustomURLPath] = useState('');
+  const [webhookParams, setWebhookParams] = useState<HeaderType[]>([
+    { key: '', value: '' },
+  ]);
+  const [webhookHeaders, setWebhookHeaders] = useState<HeaderType[]>([
+    { key: 'Content-Type', value: 'application/json' },
+  ]);
+
+  const [
+    sourceCheckboxIdToSelectedMap,
+    setSourceCheckboxIdToSelectedMap,
+  ] = useState<{
+    [x: string]: boolean;
+  }>({});
+
+  const [inputErrors, setInputErrors] = useState<CreateChannelInputErrorsType>({
+    name: false,
+    slackWebhook: false,
+    sender: false,
+    recipients: false,
+  });
+
+  useEffect(() => {
+    coreContext.chrome.setBreadcrumbs([
+      BREADCRUMBS.NOTIFICATIONS,
+      BREADCRUMBS.CHANNELS,
+      props.edit ? BREADCRUMBS.EDIT_CHANNEL : BREADCRUMBS.CREATE_CHANNEL,
+    ]);
+    window.scrollTo(0, 0);
+
+    if (props.edit) {
+      setName('test');
+      setDescription('test desc');
+      setSlackWebhook('hxxp');
+    }
+  }, []);
+
+  // returns whether inputs passed validation
+  const validateInput = (): boolean => {
+    const errors = {
+      name: validateChannelName(name),
+      slackWebhook:
+        channelType === 'SLACK' && validateSlackWebhook(slackWebhook),
```

Comment:
Np: maybe use enum/constants for Channel types

Reply:
`const channelType: "SES" | "SNS" | "SLACK" | "EMAIL" | "CHIME" | "CUSTOM_WEBHOOK"` 
tsserver already recognizes type of `channelType`, it’ll show an error if something changed. I can update it when I have the ES backend naming convention